UniWatch/Views/Class/Index.cshtml

```diff
@@ -1,54 +1,117 @@
 ﻿@model IEnumerable<UniWatch.Models.Class>
 
-<h3>My Classes</h3>
 
-<p>
-    @Html.ActionLink("New Class", "Create", null, new { @class = "btn btn-success" })
-</p>
-
-@if(!Model.Any())
+@if (Request.IsAuthenticated && User.IsInRole("Student"))
 {
-    @:It's empty in here. Create a new class.
+    <h3>My Classes</h3>
+
+    if (!Model.Any())
+    {
+        @:No class displayed. You are currently not enrolled in any class.
+    }
+    if (!Model.Any())
+    {
+        @:It's empty in here. Create a new class.
+    }
+    else
+    {
+        <div class="table-responsive">
```

Comment:
This section for the student (lines 5-61) seem to duplicate much of the code in the teacher section of the code (lines 62-116). Is there a way to minimize the duplication?